include/circt/Dialect/RTL/RTLTypes.h

```diff
@@ -19,9 +19,22 @@
 #define GET_TYPEDEF_CLASSES
 #include "circt/Dialect/RTL/RTLTypes.h.inc"
 
+namespace circt {
+namespace rtl {
+
 /// Return true if the specified type can be used as an RTL value type, that is
 /// the set of types that can be composed together to represent synthesized,
 /// hardware but not marker types like InOutType.
 bool isRTLValueType(mlir::Type type);
 
+/// Returns the type of field by name, or returns an empty type.
+mlir::Type getStructFieldType(StructType structVal, mlir::StringRef fieldName);
```

Comment:
Should/could these be a method on `StructType`?   You can add them with ` let extraClassDeclaration = [{...`